king_phisher/client/assistants/campaign.py

```diff
@@ -144,6 +220,52 @@ def is_editing_campaign(self):
 	def is_new_campaign(self):
 		return self.campaign_id is None
 
+	@property
+	def _url_thread_is_ready(self):
+		return self._url_thread is None or not self._url_thread.is_alive()
+
+	@property
+	def _server_uses_ssl(self):
+		return any(address['ssl'] for address in self.config['server_config']['server.addresses'])
+
+	def set_kpm_assistant_complete(self):
+		if not self._import_kpm.check():
```

Comment:
All four of these lines could just be `self.assistant.set_page_complete(self.assistant.get_nth_page(self.assistant.get_current_page()), self._import_kpm.check())` `KpmImport.check` needs to be moved though. If you wanted the lines to be shorter, it would be reasonable to do:
```
page = self.assistant.get_nth_page(self.assistant.get_current_page())
self.assistant.set_page_complete(page, self._import_kpm.check())
```.

Also, in general it's bad practice to invert the logic with `not` and then include an else block. Just drop the inversion and swap the blocks.